src/include/daos_obj.h

```diff
@@ -0,0 +1,786 @@
+/**
```

Comment:
It would probably be easier to review if we did the reorganization in a separate patch and then added the new object classes.   Probably not worth it at this point but would make changes like this simpler in the future.

Reply:
you are right, moving things around can be in a separate patch, I underestimated the work, I thought just a bunch of renames would be sufficient, but it took me way more time eventually.